tests/e2e/kcs/test_noobaa_rebuild.py

```diff
@@ -0,0 +1,194 @@
+import logging
+
+import pytest
+
+from ocs_ci.framework.testlib import (
+    ignore_leftovers,
+    E2ETest,
+    tier3,
+    skipif_openshift_dedicated,
+    skipif_ocs_version,
+    skipif_external_mode,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.constants import DEFAULT_NOOBAA_BUCKETCLASS, DEFAULT_NOOBAA_BACKINGSTORE
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pod import get_noobaa_pods
+from ocs_ci.ocs.resources.pvc import get_pvc_objs
+
+logger = logging.getLogger(__name__)
+
+
+@tier3
+@ignore_leftovers
+@pytest.mark.polarion_id("OCS-2653")
+@pytest.mark.bugzilla("1991361")
+@skipif_ocs_version("<4.9")
+@skipif_openshift_dedicated
+@skipif_external_mode
+class TestNoobaaRebuild(E2ETest):
+    """
+    Test to verify noobaa rebuild.
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def teardown_fixture(self, request):
+        """
+        Teardown function
+
+        """
+
+        def finalizer():
+            # Get the deployment replica count
+            deploy_obj = OCP(
+                kind=constants.DEPLOYMENT,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            noobaa_deploy_obj = deploy_obj.get(
+                resource_name=constants.NOOBAA_OPERATOR_DEPLOYMENT
+            )
+            if noobaa_deploy_obj["spec"]["replicas"] != 1:
+                logger.info(
+                    f"Scaling back {constants.NOOBAA_OPERATOR_DEPLOYMENT} deployment to replica: 1"
+                )
+                deploy_obj.exec_oc_cmd(
+                    f"scale deployment {constants.NOOBAA_OPERATOR_DEPLOYMENT} --replicas=1"
+                )
+
```

Comment:
After scaling back noobaa can we add a sanity_check to validate noobaa health, it's pods are up. What's your thought?

Reply:
I cant do it with sanity,validate_noobaa_health this method calls mcg_obj again and when i rebuild the init fails so I am validating pods/bs/bc/obc after the rebuild without this method.